onnxruntime/test/providers/cpu/tensor/grid_sample_test.cc

```diff
@@ -7,12 +7,30 @@
 namespace onnxruntime {
 namespace test {
 
-static std::unordered_set<std::string> GetExcludedExecutionProviders(int version) {
-  std::unordered_set<std::string> excluded_providers;
-  if (version >= 20) {
-    excluded_providers.insert(kCudaExecutionProvider);
+std::vector<std::unique_ptr<IExecutionProvider>> GetExecutionProviders(int opset_version) {
+  std::vector<std::unique_ptr<IExecutionProvider>> execution_providers;
+
+  execution_providers.emplace_back(DefaultCpuExecutionProvider());
+#ifdef USE_CUDA
+  if (opset_version < 20) {
+    execution_providers.emplace_back(DefaultCudaExecutionProvider());
+#ifdef ENABLE_CUDA_NHWC_OPS
+    execution_providers.push_back(DefaultCudaNHWCExecutionProvider());
+#endif
   }
-  return excluded_providers;
+
+#endif
+  return execution_providers;
+}
+
+template <typename T>
+void RunTests(T& test, std::vector<std::unique_ptr<IExecutionProvider>>& execution_providers) {
+  for (size_t idx = 0; idx < execution_providers.size(); ++idx) {
+    std::vector<std::unique_ptr<IExecutionProvider>> execution_provider;
+    execution_provider.push_back(std::move(execution_providers[idx]));
+    test.ConfigEps(std::move(execution_provider)).RunWithConfig();
```

Comment:
nit: the code can be simplified as `test.ConfigEp(execution_providers[idx]).RunWithConfig();`